nova/quota.py

```diff
@@ -892,6 +893,18 @@ def __init__(self, name, count_as_dict, flag=None, default=None):
         self.count_as_dict = count_as_dict
 
 
+class SAPCustomResource(CountableResource):
```

Comment:
Bikeshed: I'm starting to dislike the trend of the `SAP*` things. It's not descriptive.

`SAPQuotaEngine` is kind of okay, because one could say it basically says "our setup". But personally something like `DynamicQuotaEngine` (not really very good either... 🤔) would suit it better.

`SAPCustomResource` then is too generic a name for what it does for my tastes. It would better be `DedupedCountableResource` or `CheckedCountableResource` or something similar.

This wouldn't block merging for me, though. So unless we come up with a good alternative consensus, then it's fine for me to merge as-is.

Reply:
For anything we could upstream, I agree. This doesn't look upstreamable to me, so I'm fine with making clear that this is non-upstream functionality and make it easy to detect custom patches. I don't feel too strongly about that, though.

Reply:
My problem is not with upstream, but with _us_ looking at this in 10 months and not knowing what the class is about, because just because it's "SAP custom stuff" doesn't say anything about its purpose. You certainly have better recall than I do, but I'd have to re-read the code.

Reply:
fair. calling it "SAP" is the easy way to find a name, yes, but doesn't contain much. I would struggle to find a good name here, though. I'll merge as is, but will consider more rounds of name-finding for future things.